Comfyui

jdacode opened the pull request:
🛑 **New scripts must first be submitted to [ProxmoxVED](https://github.com/community-scripts/ProxmoxVED) for testing.**  
PRs for new scripts that skip this process will be closed.

---

## ✍️ Description  
Adds ComfyUI LXC Script

## 🔗 Related PR / Issue  

Link: https://github.com/community-scripts/ProxmoxVE/discussions/3915

## ✅ Prerequisites  (**X** in brackets) 

- [X] **Self-review completed** – Code follows project standards.  
- [X] **Tested thoroughly** – Changes work as expected.  
- [X] **No breaking changes** – Existing functionality remains intact.  
- [X] **No security risks** – No hardcoded secrets, unnecessary privilege escalations, or permission issues.  

---

## 🛠️ Type of Change (**X** in brackets)  

- [ ] 🐞 **Bug fix** – Resolves an issue without breaking functionality.  
- [ ] ✨ **New feature** – Adds new, non-breaking functionality.  
- [ ] 💥 **Breaking change** – Alters existing functionality in a way that may require updates.  
- [X] 🆕 **New script** – A fully functional and tested script or script set.  
- [ ] 🌍 **Website update** – Changes to website-related JSON files or metadata.  
- [ ] 🔧 **Refactoring / Code Cleanup** – Improves readability or maintainability without changing functionality.  
- [ ] 📝 **Documentation update** – Changes to `README`, `AppName.md`, `CONTRIBUTING.md`, or other docs.  

---

## 🔍 Code & Security Review  (**X** in brackets) 

- [X] **Follows `Code_Audit.md` & `CONTRIBUTING.md` guidelines**  
- [X] **Uses correct script structure (`AppName.sh`, `AppName-install.sh`, `AppName.json`)**  
- [X] **No hardcoded credentials**  


## 📋 Additional Information (optional)  
<!-- Add any extra context, screenshots, or references. -->

MickLesk commented:
Stick to the guidelines of other scripts. Not a single one has split everything into subfunctions.

We have many helper functions that simply remove many lines. Take a look at the current scripts in the live repo. It's not worth evaluating the script here, as I would basically have to criticize every line of intall.sh.

jdacode commented:
Thanks MickLesk for your prompt response. I'm happy to reduce the script. I couldn’t find anything in the guidelines about avoiding subfunctions, just curious if there's a specific reason for not using them?. Also, is there a maximum line count limit we should follow?

CrazyWolf13 commented:
@jdacode please follow the guidelines like everyone else does: https://github.com/community-scripts/ProxmoxVE/wiki/CONTRIBUTING

We want our scripts to follow the standarts so it's easier to maintain them.

jdacode commented:
I did some changes, happy to make more if needed 🙂

tremor021 commented:
Remove all comments except header ones

jdacode commented:
I made some changes to the code: 
- Removed the comments
- Added the fetch_and_deploy_gh_release function
- Ensured it only uses the latest version

@MickLesk thanks for responding. I was referring more to the case where users want to modify some parameters before bash command, for example:

`comfyui_python_version_uv="3.12" bash -c "$(curl -fsSL https://.../ct/comfyui.sh)"`

I think this is super useful. However, I'm happy to remove the dynamic variables from the script if you guys want

jdacode commented:
Hey @michelroegl-brunner thanks for your input.

I either removed the variable or changed it to a local one. I also removed or shortened some of the dialogs. Let me know if theres anything else to change.

jdacode commented on install/comfyui-install.sh:
```diff
@@ -0,0 +1,109 @@
+
+msg_info "Install ${application_name} Manager"
+comfyui_manager_dir="${comfyui_path}/custom_nodes/comfyui-manager"
+git clone https://github.com/ltdrdata/ComfyUI-Manager "${comfyui_manager_dir}"
```
I removed the ComfyUI Manager. The project doesn't have releases (only tags), so I can't use fetch_and_deploy_gh_release. I also tried downloading the tar package provided by the project, but it doesn't work with the latest ComfyUI. So, I decided to remove the manager, and if someone needs it, it's better to install it manually.

jdacode commented on install/comfyui-install.sh:
```diff
@@ -0,0 +1,109 @@
+[Install]
+WantedBy=multi-user.target
+EOF
+systemctl enable -q --now "${application_name}"
```
ok done

jdacode commented on install/comfyui-install.sh:
```diff
@@ -0,0 +1,86 @@
+
+msg_info "Cleaning up"
+$STD apt-get -y autoremove
+$STD apt-get -y autoclean
```
Oh thanks great to know. Changed to apt.
